cmd/kubeadm/app/apis/output/types.go

```diff
@@ -82,3 +82,37 @@ type UpgradePlan struct {
 
 	ConfigVersions []ComponentConfigVersionState
 }
+
+// Certificate represents information for a certificate or a certificate authority when using the check-expiration command.
+type Certificate struct {
+	// Name of the certificate.
+	Name string
+
+	// ExpirationDate defines certificate expiration date.
+	ExpirationDate metav1.Time
+
+	// ResidualTime represents the time missing to expiration.
```

Comment:
```suggestion
	// ResidualTime represents the residual time before expiration.
```

residual time = remaining time